### PR TITLE
Timestamp (annotate with a timestamp) all messages automatically

### PR DESCRIPTION
**Pull request by dcorbacho**
## Types of Changes
- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist
- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

File: deps/rabbit/src/rabbit_priority_queue.erl [hunk 1]
```diff
@@ -689,6 +691,28 @@ find_head_message_timestamp(BQ, [{_, BQSN} | Rest], Timestamp) ->
...
+    Timestamps =
+        lists:foldl(
+          fun({_, BQSN}, Acc) ->
+                  case oldest_message_received_timestamp(BQ, BQSN) of
+                      '' -> Acc;
+                      Ts -> [Ts | Acc]
+                  end
+          end, [], BQs),
+    case Timestamps of
+        [] -> '';
```
Comment (michaelklishin) [inline deps/rabbit/src/rabbit_priority_queue.erl:705]: Should we return `undefined` and not an empty atom when the queue has no messages?

**Review by michaelklishin — CHANGES_REQUESTED** [review]

**Comment by michaelklishin** [comment]
The force-push was a rebase.

File: deps/rabbit/src/rabbit_priority_queue.erl (hunk 1 above)
Comment (dcorbacho) [reply deps/rabbit/src/rabbit_priority_queue.erl:705]: It returns an empty atom to follow the same format as `head_message_timestamp` and any other empty values returned by `rabbit_amqqueue:info`.

Comment (michaelklishin) [reply deps/rabbit/src/rabbit_priority_queue.erl:705]: OK, I have suspected it might have been for consistency with certain decisions from years ago.

**Review by michaelklishin — APPROVED** [review]

**Pull request merged**